pom.xml

```diff
@@ -1331,6 +1336,28 @@
             </exclusions>
           </dependency>
           <dependency>
+          	<groupId>org.apache.avro</groupId>
+          	<artifactId>avro</artifactId>
+          	<version>1.7.4</version>
```

Comment:
hive-exec-shade includes Avro 1.7.5. java-exec uses 1.7.7. Hive-exec uses ${avro.version} -- which is not set anywhere that I can find.

The key here is to exclude snappy, so I moved the exclusion into java-exec.